src/sinks/prometheus/exporter.rs

```diff
@@ -160,39 +179,160 @@ impl SinkConfig for PrometheusCompatConfig {
+// the type (counter, gauge, etc) and any subtype information like histogram buckets.
+//
+// Specifically, though, we do _not_ include the actual metric value.  This type is used
+// specifically to look up the entry in a map for a metric in the sense of "get the metric whose
+// name is X and type is Y and has these tags".
+#[derive(Clone, Debug)]
+struct MetricRef(
+    MetricSeries,
+    MetricKind,
+    Discriminant<MetricValue>,
+    Option<Vec<f64>>,
+);
+
+impl MetricRef {
+    /// Creates a `MetricRef` based on the given `Metric`.
+    pub fn from_metric(metric: &Metric) -> Self {
+        // Either the buckets for an aggregated histogram, or the quantiles for an aggregated summary.
+        let bounds = match metric.value() {
+            MetricValue::AggregatedHistogram { buckets, .. } => {
+                Some(buckets.iter().map(|b| b.upper_limit).collect())
+            }
+            MetricValue::AggregatedSummary { quantiles, .. } => {
+                Some(quantiles.iter().map(|q| q.quantile).collect())
+            }
+            _ => None,
+        };
+
+        Self(
+            metric.series().clone(),
+            metric.kind(),
+            discriminant(metric.value()),
+            bounds,
+        )
+    }
+}
+
+impl PartialEq for MetricRef {
+    fn eq(&self, other: &Self) -> bool {
+        self.0 == other.0 && self.1 == other.1 && self.2 == other.2 && self.3 == other.3
+    }
+}
+
+impl Eq for MetricRef {}
+
+impl Hash for MetricRef {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.0.hash(state);
+        self.1.hash(state);
+        self.2.hash(state);
+        if let Some(bounds) = &self.3 {
+            for bound in bounds {
+                bound.to_bits().hash(state);
+            }
+        }
+    }
+}
+
+struct PrometheusExporterMetricNormalizer {
+    distributions_as_summaries: bool,
+    buckets: Vec<f64>,
+}
+
+impl MetricNormalize for PrometheusExporterMetricNormalizer {
+    fn apply_state(&mut self, state: &mut MetricSet, metric: Metric) -> Option<Metric> {
+        let new_metric = match metric.value() {
+            MetricValue::Distribution { .. } => {
+                // Convert the distribution as-is, and then let the normalizer absolute-ify it.
+                let (series, data, metadata) = metric.into_parts();
+                let (ts, kind, value) = data.into_parts();
+
+                let new_value = match value {
+                    MetricValue::Distribution { samples, .. } => {
+                        if self.distributions_as_summaries {
+                            // We use a sketch when in summary mode because they're actually able to be
+                            // merged and provide correct output, unlike the aggregated summaries that
+                            // we handle from _sources_ like Prometheus.  The collector code already
+                            // renders sketches as an aggregated summary to begin with, so we're gucci.
```

Comment:
"we're gucci"?

Reply:
There I go again, treating code comments like Slack. :D

I've gone ahead and made the language a bit more plain/straightforward.